src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java

```diff
@@ -379,8 +383,10 @@ private IrisSubSettings copyIrisSubSettings(IrisSubSettings target, IrisSubSetti
         if (target == null || source == null) {
             return source;
         }
-        target.setEnabled(source.isEnabled());
-        target.setPreferredModel(source.getPreferredModel());
+        if (authCheckService.isAdmin()) {
+            target.setEnabled(source.isEnabled());
+            target.setPreferredModel(source.getPreferredModel());
+        }
```

Comment:
Should we throw an exception when the user tries to change these values but does not have the rights to do so?

Reply:
I don't think this is needed here. The UI will automatically switch back to the old values after saving, so this should indicate it to the user.
This is only a temporary and short-term fix, as I'm already working on reworking the settings system. As part of this rework, I will also handle this case properly.

However I now also disabled the fields in the UI if the user is not an admin, so this case can't even really happen.